src/Unsafe/MutableArray.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE MagicHash #-}
+{-# LANGUAGE UnboxedTuples #-}
+
+-- |
+-- Module: MutableArray
+-- Description: Unsafe wrappers around `MutableArray#`s
+module Unsafe.MutableArray
+  ( newMutArr,
+    readMutArr,
+    writeMutArr,
+  )
+where
+
+import GHC.Exts
+
+-- # Unsafe wrappers
+----------------------------
+
+newMutArr :: Int -> a -> MutableArray# RealWorld a
+newMutArr (I# size) x =
+  case newArray of
+    (# _, array #) -> array
+  where
+    newArray = runRW# $ \stateRW -> newArray# size x stateRW
+
+-- | writeMutArr should have evaluation forced
```

Comment:
I still don't understand what this means.

Reply:
Forgot to change it. Will when I merge.